.circleci/config.yml

```diff
@@ -47,33 +47,21 @@ aliases:
       set -u
       set -o pipefail
 
-      # This Shallow Git Clone code is adapted from what the standard CircleCI `checkout` command does for the case of an external PR (link to example below):
-      # https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/49817/workflows/dc195ea6-ac06-4de1-9edf-4c949427b5fb/jobs/1430976/parallel-runs/0/steps/0-101
-      ### git clone --no-checkout "$CIRCLE_REPOSITORY_URL" .
-      ### git fetch --force origin +refs/pull/18748/head:refs/remotes/origin/pull/18748
-      ### git checkout --force -B "$CIRCLE_BRANCH" "$CIRCLE_SHA1"
-      ### git --no-pager log --no-color -n 1 --format='HEAD is now at %h %s'
-
       # Set up SSH access
       # This SSH key is the current github.com SSH key as of June 2023, but it will need to be changed whenever github changes their key (probably every few years)
       GITHUB_SSH_KEY="AAAAC3NzaC1lZDI1NTE5AAAAIOMqqnkVzrm0SdG6UOoqKLsabgH5C9okWi0dh2l9GKJl"
       mkdir -p ~/.ssh
       echo github.com ssh-ed25519 $GITHUB_SSH_KEY >> ~/.ssh/known_hosts
 
```

Comment:
What if the `$CIRCLE_SHA1` in `git fetch --depth 1 origin "$CIRCLE_SHA1"` is not the hash of the commit at depth 1 in the `git clone --depth 1 --no-checkout "$CIRCLE_REPOSITORY_URL" .` command?

Reply:
The commit used for the checkout doesn't matter. We just need _some_ commit to be the current one, until we checkout the correct commit.

All that the fetch command relies on is that the given hash is on the remote, which is the case (even for forks, as long as they have created a PR to our repo).

Reply:
ah, okay, that makes sense